sources/logdog/src/log_request.rs

```diff
@@ -166,24 +164,73 @@ where
     Ok(())
 }
 
+pub(crate) fn normalize(
+    json: &serde_json::Value,
+    prefix: String,
+    flattened: &mut HashMap<String, serde_json::Value>,
+) -> Result<()> {
+    match json {
+        serde_json::Value::Object(map) => {
+            for (k, v) in map {
+                let new_key = if prefix.is_empty() {
+                    k.clone()
+                } else {
+                    format!("{}.{}", prefix, k)
+                };
+                normalize(v, new_key, flattened)?;
+            }
```

Comment:
The use of `.` as a separator is ambiguous because some keys have dots in them. `settings.kernel.sysctl` is one but there are many others. This approach would corrupt the structure when unflattened.

I would look instead at `serde_json::Value` and [pointer_mut](https://docs.rs/serde_json/latest/serde_json/value/enum.Value.html#method.pointer_mut) which takes a JSONPointer query. Then you could replace the value with "REDACTED" if it is a string.